### ISSUE TITLE
Possible to automate restart of console/terminal after 1 1/2 hours?

### ISSUE THREAD

**Issue by pedda007** [jabbink/PokemonGoBot#843]
hi!
just wanted to aski if it is possible to restart automatically the terminal or console after 1 1/2 hours?
Anybody tried something like that?

Thanks!

**Comment by chemikaz** [comment]
Hi, I'm using this bash code:

#!/bin/bash
while [ 1 ]; do
killall java;
java -jar PokemonGoBot-0.5.0.jar & sleep 5400;
done

**Comment by russr123** [comment]
only issue you will have is on the restart, it will teleport to the start, if thats far from the end when it shut off, then it will soft ban 

i use this as a auto off timer.. 

make a bat script to start the bot, called run.bat, use this as the timer, you need to start the bot with this script for it to work..

start "pogo" run.bat
timeout /t 2400
taskkill /fi "WINDOWTITLE eq pogo*" /f
taskkill /fi "WINDOWTITLE eq pogo*" /f
taskkill /fi "WINDOWTITLE eq pogo*" /f

the number is seconds, 2400 = 40min

i use 3 taskkills because when i used 1 it didnt always kill it...

**Comment by chemikaz** [comment]
That's true if you are on mac and use my script you can add: "sleep 600;" after "killall java;"

**Comment by pedda007** [comment]
Thanks guys for your answers. I'm on mac and i have completely no idea, how i do it. I'm really sorry for such newbie questions. But how do i do that things? Do i have to make an extra file for that? Or just to type that things in the terminal? 
Thanks

**Comment by chemikaz** [comment]
Create new file "script.sh" in same directory where is jar file. Copy this lines in to that file:

#!/bin/bash
while [ 1 ]; do
killall java;
java -jar PokemonGoBot-0.5.0.jar & sleep 5400;
sleep 600;
done

Make your file executable chmod +x script.sh
Run your script: ./script.sh

**Comment by pedda007** [comment]
Sorry guys, i dont get it.
I wrote these exact lines in the text edit and saved it as script2.sh.
I add a screenshot where you see the script file and then the terminal "output", or how you call it xD
Maybe you can help me out one more time :)

thaaaanks
![bildschirmfoto 2016-08-08 um 16 26 49](https://cloud.githubusercontent.com/assets/20835895/17483247/91e4d9e4-5d85-11e6-89db-eb598f022cbb.png)

**Comment by urbalazs** [comment]
My solution is cron + bash script:

$ crontab -l
```
# m h  dom mon dow   command
0 */2 * * * bash /home/urbalazs/bot-restart.sh
```

$ cat bot-restart.sh
```
#!/bin/bash

killall java
sleep 10
cd /home/urbalazs/PokemonGoBot
java -jar PokemonGoBot-0.5.0.jar --spring.main.web-environment=false >> log-`date +'%Y%m%d'`.txt &
```

The bot logs its activity to a file. It creates a new file every day, like: `log-20160808.txt`

**Comment by vamirineni87** [comment]
This is my script.
2 batch files - 

run.bat
```
@echo off
setlocal enableextensions disabledelayedexpansion
TITLE pogoscrapper-controller

:loop
set "now=%time: =0%"
echo %now%
set "task=runbot"
if "%now%" geq "20:00:00,00" ( set "task=idlebot" )
if "%now%" lss "07:00:00,00" ( set "task=idlebot" )
echo Task is %task%

SET /A randvar=%RANDOM%
echo Random var - %randvar%
set thrushold=15000
if %randvar% geq %thrushold% ( 
echo "skipping"
goto task_idlebot
)
call :task_%task%

:task_runbot
start /min runpogo.bat
SET /A seconds=%RANDOM% * 1200 / 32768 + 1200
timeout /t %seconds%
taskkill /f /FI "WINDOWTITLE eq running-pogoscrapper" /T

:task_idlebot
SET /A seconds=%RANDOM% * 600 / 32768 + 300
timeout /t %seconds%
goto loop

```

runpogo.bat
```
TITLE running-pogoscrapper
"C:\Program Files\Java\jre1.8.0_101\bin\java.exe" -jar PokemonGoBot-0.5.0.jar
pause
```

**Comment by chemikaz** [comment]
pedda007, have you added last line where is word "done" ?

**Comment by pedda007** [comment]
i added this word in this second.... now it WORKS!!!! Thank you very much!
But when does the relog come? In 10 minutes? Because i waited 10min and it didn't restart...

**Comment by chemikaz** [comment]
Its will be running for 90min then pausing for 10min and restarts;]

**Comment by pet90210** [comment]
@chemikaz  how can I bring this to work when I'm using the "source" and not the prebuild on OSX?

`#!/bin/bash
while [ 1 ]; do
./gradlew run & sleep 5400;
killall java;
sleep 600;`
?

**Comment by chemikaz** [comment]
I think it should work. Let me know after 1,5hr :)

**Comment by pet90210** [comment]
@chemikaz - I'll report ;)

**Comment by zhoushifeng8** [comment]
hi @chemikaz after 1.5 hours it not works

**Comment by chemikaz** [comment]
@zhoushifeng8 should be 10min pause before restart.

**Comment by zhoushifeng8** [comment]
@chemikaz wait 30mins and not restart...

**Comment by pet90210** [comment]
didn't work either ... could not kill the javaprocess with one time :/

**Comment by yscphdk** [comment]
Can someone make a video tutorial for us rookies?

**Comment by BugSyst3m** [comment]
urbalazs, It works great, I simply change the sleep time to 600 to avoid softbans.

**Comment by pet90210** [comment]
okay ... works for me on mac. 
Killing it 3 times like @russr123 

```
#!/bin/bash
while [ 1 ]; do
./gradlew run & sleep 5400;
killall java;
killall java;
killall java;
sleep 600;
done
`

**Comment by pedda007** [comment]
thanks chemikaz! Worked out great for me!!! Unfortunately, i got to much Pokemon in my inventory todays morning xD

**Comment by dominic676767** [comment]
@chemikaz 

I tried running your script like this 

#!/bin/bash
while [ 1 ]; do
java -jar PokemonGoBot-0.5.0.jar & sleep 5400;
killall java;
sleep 600;
done

but it wont work on my Mac terminal.

I tried to resolve the problem by modifying the script a lil bit due to permission issue on my Mac and run it in Terminal using chmod +x script.sh
The script goes like this

#!/bin/bash
while [ 1 ]; do
cd /Users/Dominic676767/Downloads/PokemonGoBot
java -jar PokemonGoBot-0.5.1.jar & sleep 5400;
killall java;
sleep 600;
done

The java file just wont launch. Any solutions on this issue ?
Thanks.

**Comment by chemikaz** [comment]
@dominic676767, you don't need to use any extra scripts with 0.5.1 version. Working fine after 1,5hr;]

**Comment by dominic676767** [comment]
@chemikaz , thanks a lot ! I wasn't aware that was implemented with the 0.5.1 version.
